modules/storage-api/src/main/java/org/apache/ignite/internal/storage/util/StorageOperation.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage.util;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Storage operations.
+ */
+interface StorageOperation {
+    /**
+     * Storage creation operation.
+     */
+    class CreateStorageOperation implements StorageOperation {
+    }
+
+    /**
+     * Storage destruction operation.
+     */
+    class DestroyStorageOperation implements StorageOperation {
+        private final CompletableFuture<Void> destroyFuture = new CompletableFuture<>();
+
+        private final AtomicReference<CreateStorageOperation> createStorageOperationReference = new AtomicReference<>();
```

Comment:
Looks like this can be just AtomicBoolean

Reply:
Not suitable, I need to atomically replace it with the creation operation when the destruction operation is completed if necessary.